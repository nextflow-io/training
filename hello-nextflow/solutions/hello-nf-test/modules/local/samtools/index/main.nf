/*
 * Generate BAM index file
 */
process SAMTOOLS_INDEX {

    container 'quay.io/biocontainers/samtools:1.19.2--h50ea8bc_1' 

    input:
        tuple val(id), path(input_bam)

    output:
        tuple val(id), path(input_bam), path("${input_bam}.bai")

    """
    samtools index '$input_bam'

    """
}
