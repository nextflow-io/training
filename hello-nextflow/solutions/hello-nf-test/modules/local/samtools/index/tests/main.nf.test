nextflow_process {

    name "Test Process SAMTOOLS_INDEX"
    script "../main.nf"
    process "SAMTOOLS_INDEX"

    test("reads_son [bam]") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = file("${projectDir}/data/bam/reads_son.bam")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
    test("reads_mother [bam]") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = file("${projectDir}/data/bam/reads_mother.bam")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
    test("reads_father [bam]") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = file("${projectDir}/data/bam/reads_father.bam")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
