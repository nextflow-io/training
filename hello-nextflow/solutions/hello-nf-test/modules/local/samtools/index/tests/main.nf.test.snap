{
    "reads_father [bam]": {
        "content": [
            {
                "0": [
                    [
                        "reads_father.bam:md5,57f3b977b0eee8278924e38894d02a57",
                        "reads_father.bam.bai:md5,49cb8454781a279a8b33f47209e76fe5"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-10-27T23:45:04.526628608"
    },
    "reads_son [bam]": {
        "content": [
            {
                "0": [
                    [
                        "reads_son.bam:md5,af5956d9388ba017944bef276b71d809",
                        "reads_son.bam.bai:md5,a2ca7b84998218ee77eff14af8eb8ca2"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-10-27T23:26:47.056835774"
    },
    "reads_mother [bam]": {
        "content": [
            {
                "0": [
                    [
                        "reads_mother.bam:md5,04b918c3707c24d8f3a7e88bb68e2d32",
                        "reads_mother.bam.bai:md5,1b17cbcc9392e7e787cc83a9081deb31"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-10-27T23:44:54.193439354"
    }
}